assets/dev/js/editor/editor-base.js

```diff
@@ -287,7 +287,8 @@ export default class EditorBase extends Marionette.Application {
 			}
 
 			if ( ! this.widgetsCache[ widgetType ].commonMerged && ! this.widgetsCache[ widgetType ].atomic_controls ) {
-				jQuery.extend( this.widgetsCache[ widgetType ].controls, this.widgetsCache.common.controls );
+				const commonControls = elementor.hooks.applyFilters( 'elements/widget/controls/common/default', this.widgetsCache.common.controls, widgetType );
```

Comment:
@nuritsha , I would be interested to see the use case for this :) But anyway, I think this filter might be too early. In the "Optimized Markup" experiment, we introduced a new Advanced tab `common-optimized`, so you might need to add the filter after that. Maybe just double check?

Reply:
@louiswol94 I don't see any reference to that experiment in the js code, can you please direct me to the relevant place?

Reply:
@nuritsha The experiment name is `e_optimized_markup`. You will see on [line 302](https://github.com/elementor/elementor/blob/e321132bc41340a2a4af4012aabaa8ec85e6afdc/assets/dev/js/editor/editor-base.js#L302) we merge those controls in as well. So I am thinking the filter should only be added after that?

Reply:
yep, u r right